gschem/src/x_newtext.c

```diff
@@ -75,7 +75,8 @@ dialog_response_apply (NewText *dialog)
   int align = LOWER_LEFT;
   int color = TEXT_COLOR;
   int rotate = 0;
-  int size = 12;
+  GschemToplevel *w_current = GSCHEM_DIALOG (dialog)->w_current;
```

Comment:
Please avoid using of direct links to data structures with *gobjects*. They have better means.
Use `GSCHEM_DIALOG`'s property to get `w_current`. E.g.
>  GschemToplevel *w_current;
>  int size;
>   g_return_if_fail (dialog != NULL);
>  g_object_get (GSCHEM_DIALOG (dialog), "gschem-toplevel", &w_current, NULL);
>  g_return_if_fail (w_current != NULL);
>  size = w_current->text_size;

Thus you can check inappropriate usage of the *dialog_response_apply()* on the build stage.
If you're planning to somehow use this dialog without `w_current`, you should check it and exit if it is `NULL` (I don't think it is good idea just now).

In future it would be better to use `gschem_options` to store various settings, I think.